cmd/swarmd/main.go

```diff
@@ -126,6 +128,12 @@ var (
 				return err
 			}
 
+			resources := &api.Resources{ScalarResources: map[string]float64{}}
+			err = flagparser.ParseUserDefinedResources(cmd.Flags(), resources, "resources", false)
```

Comment:
From other example, this place only reads parameter in built-in types. The parse and validation is done separately.